keps/976-plain-pods/README.md

```diff
@@ -0,0 +1,669 @@
+
+<<[UNRESOLVED usability of labels and annotations ]>>
+Could the split among labels and annotations cause user mistakes?
+Note that integers cannot be used as label values.
+<<[/UNRESOLVED]>>
+
+The pods of a job with a single spec (similar to Pods in an Indexed Job), look like follows:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: pod-index-0
+  namespace: pod-namespace
+  labels:
+    kueue.x-k8s.io/queue-name: user-queue
+    kueue.x-k8s.io/pod-group-name: pod-group
+  annotations:
+    kueue.x-k8s.io/pod-group-count: "10"
+spec:
+  containers:
+  - name: job
+    image: hello-world
+    resources:
+      requests:
+        cpu: 1m
+```
+
+#### Story 3
+
+As a platform developer, I can queue groups of Pods that have multiple shapes.
+In addition to the queue name, I can specify how many shapes to expect for the group and how
+many Pods are expected for the shape.
+
+The pods of a job following a driver-workers paradigm look like follows:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: job-driver
+  labels:
+    kueue.x-k8s.io/queue-name: user-queue
+    kueue.x-k8s.io/pod-group-name: pod-group
+    kueue.x-k8s.io/pod-group-role: driver
+  annotations:
+    kueue.x-k8s.io/pod-group-roles: "2"
+    kueue.x-k8s.io/pod-group-role-count: "1"
+spec:
+  containers:
+  - name: job
+    image: hello-world
+    resources:
+      requests:
+        cpu: 1m
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: job-worker-0
+  labels:
+    kueue.x-k8s.io/queue-name: user-queue
+    kueue.x-k8s.io/pod-group-name: pod-group
+    kueue.x-k8s.io/pod-group-role: worker
+  annotations:
+    kueue.x-k8s.io/pod-group-roles: "2"
+    kueue.x-k8s.io/pod-group-role-count: "10"
+spec:
+  containers:
+  - name: job
+    image: hello-world
+    resources:
+      requests:
+        cpu: 1m
+```
+
+### Story 4
+
+Motivation: In frameworks like Spark, worker Pods are only created after by the driver Pod. As
+such, the worker Pods specs cannot be predicted beforehand. Even though the job could be considered
+elastic, generally users wouldn't want to start a spark driver to run if no workers would fit.
+
+As a Spark user, I can queue the driver Pod while providing the expected shape of the worker Pods.
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: job-driver
+  labels:
+    kueue.x-k8s.io/queue-name: user-queue
+    kueue.x-k8s.io/pod-group-name: pod-group
+    kueue.x-k8s.io/pod-group-role: driver
+  annotations:
+    kueue.x-k8s.io/pod-group-role-count: "1" # optional
+    # The template in the driver group can be left empty. Kueue will populate it from the Pod.
+    kueue.x-k8s.io/pod-group-sets: |-
+      [
+        {
+          name: driver,
+          count: 1,
+        },
+        {
+          name: workers,
+          count: 10,
+          template:
+            spec:
+              containers:
+                - name: worker
+                  requests:
+                    cpu: 1m
+        }
+      ]
```

Comment:
I think that if we allow users to set unlimited strings here, there is a potential issue with the size of the annotation. IIRC, the annotations size limit is 256KiB.

So, can we limit the size of ` kueue.x-k8s.io/pod-group-sets` or mention the potential issue in `Risks and Mitigations` section?

Reply:
Let me add to risks